### ISSUE TITLE
unhandled exception : M30_PT1/Grunt_NPC_Pistol_6 -> NPC_Logic_1 -> NPC_LOGIC_DUDE

### ISSUE THREAD

**Issue by ghramsey** [MattFiler/OpenCAGE#315]
I found this and initially thought it was related to my renaming a character, but it also happened on the 
EGS version where I have a vanilla, unaltered game installed.

I'm going to try to describe how to reproduce this, but I'm not 100% sure I'm writing it correctly.

This is an alias for one of the NPCs in M16.  I clicked the alias instance
 Grunt_NPC_Pistol_6 -> NPC_Logic_1 -> NPC_LOGIC_DUDE in the M30_PT01 composite and it 
 opened the alias instance. I used the GO button to open the NPC_LOGIC_DUDE linked there to.
 
I returned to the M30_PT1 composite and happened to go through those same steps a second time which 
throws the unhandled exception.  'Continue' works and I see no further errors

I found that clicking on the alias in the list stop the error.  
If I don't  then using the go button causes the exception.

I am uncertain where the exact issue is so I've made a somewhat vague title above.

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at CATHODE.Scripting.EntityUtils.GetName(Composite composite, Entity entity)
   at CommandsEditor.DockPanels.EntityDisplay.Reload()
   at CommandsEditor.DockPanels.EntityDisplay..ctor(CompositeDisplay compositeDisplay, Entity entity)
   at CommandsEditor.DockPanels.CompositeDisplay.LoadEntity(Entity entity)
   at CommandsEditor.DockPanels.CompositeDisplay.composite_content_SelectedIndexChanged(Object sender, EventArgs e)
   at System.Windows.Forms.ListView.OnSelectedIndexChanged(EventArgs e)
   at System.Windows.Forms.ListView.WmReflectNotify(Message& m)
   at System.Windows.Forms.ListView.WndProc(Message& m)
   at ListViewGroupCollapse.ListViewExtended.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9181.0 built by: NET481REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
CommandsEditor
    Assembly Version: 0.2.0.0
    Win32 Version: 0.2.0.0
    CodeBase: file:///D:/Program%20Files/Epic%20Games/AlienIsolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9172.0 built by: NET481REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9181.0 built by: NET481REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9037.0 built by: NET481REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
WeifenLuo.WinFormsUI.Docking
    Assembly Version: 1.0.0.0
    Win32 Version: 0.2.0.0
    CodeBase: file:///D:/Program%20Files/Epic%20Games/AlienIsolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
DiscordRPC
    Assembly Version: 1.2.1.24
    Win32 Version: 0.2.0.0
    CodeBase: file:///D:/Program%20Files/Epic%20Games/AlienIsolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
CathodeLib
    Assembly Version: 0.5.1.0
    Win32 Version: 0.2.0.0
    CodeBase: file:///D:/Program%20Files/Epic%20Games/AlienIsolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
netstandard
    Assembly Version: 2.0.0.0
    Win32 Version: 4.8.9037.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/netstandard/v4.0_2.0.0.0__cc7b13ffcd2ddd51/netstandard.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9200.0 built by: NET481REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9037.0 built by: NET481REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9037.0 built by: NET481REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
PresentationCore
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9200.0 built by: NET481REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/PresentationCore/v4.0_4.0.0.0__31bf3856ad364e35/PresentationCore.dll
----------------------------------------
WindowsBase
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9200.0 built by: NET481REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/WindowsBase/v4.0_4.0.0.0__31bf3856ad364e35/WindowsBase.dll
----------------------------------------
Newtonsoft.Json
    Assembly Version: 13.0.0.0
    Win32 Version: 0.2.0.0
    CodeBase: file:///D:/Program%20Files/Epic%20Games/AlienIsolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
WeifenLuo.WinFormsUI.Docking.ThemeVS2015
    Assembly Version: 1.0.0.0
    Win32 Version: 0.2.0.0
    CodeBase: file:///D:/Program%20Files/Epic%20Games/AlienIsolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
System.Xml.Linq
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9037.0 built by: NET481REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml.Linq/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.Linq.dll
----------------------------------------
System.Runtime.Serialization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9037.0 built by: NET481REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9037.0 built by: NET481REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.9191.0 built by: NET481REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
websocket-sharp
    Assembly Version: 1.0.2.19803
    Win32 Version: 0.2.0.0
    CodeBase: file:///D:/Program%20Files/Epic%20Games/AlienIsolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
System.Numerics.Vectors
    Assembly Version: 4.1.4.0
    Win32 Version: 0.2.0.0
    CodeBase: file:///D:/Program%20Files/Epic%20Games/AlienIsolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------
System.Memory
    Assembly Version: 4.0.1.1
    Win32 Version: 0.2.0.0
    CodeBase: file:///D:/Program%20Files/Epic%20Games/AlienIsolation/DATA/MODTOOLS/REMOTE_ASSETS/scripteditor/CommandsEditor.exe
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
```

**Comment by MattFiler** [comment]
I've just been trying to follow your reproduction steps here and am a bit confused. You mention M16, but say you're editing M30_PT01 on ENG_TOWPLATFORM? I can't find the Grunt NPC in the M30 composite that you're referring to.

**Comment by ghramsey** [comment]
I'm sorry for the confusion.  I wrote "M16" so I think I meant "Mission 16" which is M30_PT01 in the Tech_Comms file scripts.

In the M30_PT01 story mission there's a lone alias entity. 
![image](https://github.com/MattFiler/OpenCAGE/assets/51859157/fc591170-acb2-4961-a28c-195673df1593)


I think we could consider this fixed though. It might be related to the other bugs that you've just fixed.
Following the steps I wrote I cannot reproduce it at this stage either.
